src/DotNetLightning.Core/Transactions/Transactions.fs

```diff
@@ -348,7 +353,7 @@ module Transactions =
                 remotePaymentBasepoint
 
     /// Sort by BOLT 3: Compliant way (i.e. BIP69 + CLTV order)
-    let sortTxOut (txOutsWithMeta: (TxOut * _) list) =
+    let sortTxOut (txOutsWithMeta: List<TxOut * Option<UpdateAddHTLCMsg>>) =
```

Comment:
nit: I prefer `'T list` style to avoid confusion with `ResizeArray`

Reply:
we'll change it, but I'm curious why would `List<'T>` be confused with ResizeArray?

Reply:
Because `ResizeArray<T>` is just an alias for `System.Collections.Generic.List<T>`.
`List<T>` is ambiguous, we can not tell if it is `FSharp.Colletions.List<T>` or `System.Collections.Generic.List<T>`
`T list` is always `FSharp.Collections.List<T>`

Reply:
Not sure I agree with that take (I'll give my rationale later), but why `'T list` instead of `list<'T>`?

Reply:
> but why 'T list instead of list<'T>?

No big reason for this, just because it feels natural to post-fix the type parameter with the wrapper type in case the wrapper type is lowercase.
i. e.
`'T option`, `Option<'T>` feels more natural than `option<'T>` , `'T Option`

but again, it is not a big deal.

Reply:
Ok, this is my take (which you might not agree with):

- I generally prefer APIs to follow .NET design guidelines, which means PascalCase for methods, type names, etc. Using lowercase for type names or methods, like some F#-ers do, strikes to me as very unwelcoming to C# people. I know in this case the distinction between list and List is useful because one guarantees immutability, but IMO the API consistency with C# is more important (and you can still enforce immutability via not allowing to open System.Collections.Generic at review time).
- Writing sometimes `'Foo Bar` and other times `Bar<'Foo>` is inconsistent to me. Most F# documentation advocates for the latter format so I always try to use the latter, which also looks less ambiguous and confuses C# devs less.

(My motivation here is mainly the thought that the group of C# devs that might join us into the F# club can be much bigger than the group of FP devs that might come from other langs. This is also why I don't use paket myself, but just nuget, etc.)